metapool/sample_sheet.py

```diff
@@ -280,24 +280,34 @@ def write_blank_lines(writer, n=blank_lines, width=csv_width):
     def merge(self, sheets):
         """Merge the Data section of multiple sample sheets
 
+        For the Date field in the Header section, we only keep the date of the
+        current sheet.
+
         Parameters
         ----------
         sheets: list of KLSampleSheet
             The sample sheets to merge into `self`.
 
-        Returns
-        -------
-        KLSampleSheet
-            The combined sample sheet.
-
         Raises
         ------
         ValueError
-            If the header section is different between merged sheets.
+            If the Header, Settings or Reads section is different between
+            merged sheets.
         """
         for number, sheet in enumerate(sheets):
             for section in ['Header', 'Settings', 'Reads']:
-                if getattr(self, section) != getattr(sheet, section):
+
+                # For the Header section we'll ignore the Date field since that
+                # is likely to be different but shouldn't be a condition to
+                # merge two sheets.
+                this, that = getattr(self, section), getattr(sheet, section)
```

Comment:
If this is only needed if sections is == 'Header', should it be under the if?

Reply:
My bad, that comment section was in the wrong location. It is still handy to have `this` and `that`, but the Date check is specific for the Header section. I've reordered this, should be clearer now.